agent/api/task.go

```diff
@@ -867,6 +867,7 @@ func (task *Task) UpdateStatus() bool {
 func (task *Task) UpdateDesiredStatus() {
 	task.updateTaskDesiredStatus()
 	task.updateContainerDesiredStatus()
+	task.updateResourceDesiredStatus()
```

Comment:
I agree this is not the right behavior. We are probably looking at an already existing issue here. 
We should probably hold the task lock for all the three update*Status() here and have a new `setDesiredStatusUnsafe` so that we can set task's desired status safely. Does that sound good?

Reply:
that should work, but I'm not sure if `setDesiredStatusUnsafe` is required, you can just reference it in these updates methods, since it's already protected by a lock.